src/commands/images/pushImage.ts

```diff
@@ -5,48 +5,69 @@
 
 import vscode = require('vscode');
 import { IActionContext } from 'vscode-azureextensionui';
-import { configurationKeys } from '../../constants';
 import { ext } from '../../extensionVariables';
 import { localize } from '../../localize';
 import { ImageTreeItem } from '../../tree/images/ImageTreeItem';
-import { askToSaveRegistryPath } from '../registries/registrySettings';
+import { registryExpectedContextValues } from '../../tree/registries/registryContextValues';
+import { RegistryTreeItemBase } from '../../tree/registries/RegistryTreeItemBase';
 import { addImageTaggingTelemetry, tagImage } from './tagImage';
 
 export async function pushImage(context: IActionContext, node: ImageTreeItem | undefined): Promise<void> {
     if (!node) {
         node = await ext.imagesTree.showTreeItemPicker<ImageTreeItem>(ImageTreeItem.contextValue, {
             ...context,
-            noItemFoundErrorMessage: localize('vscode-docker.commands.images.push.noImages', 'No images are available to push')
+            noItemFoundErrorMessage: localize('vscode-docker.commands.images.push.noImages', 'No images are available to push'),
         });
     }
 
-    const defaultRegistryPath = vscode.workspace.getConfiguration('docker').get(configurationKeys.defaultRegistryPath);
+    let connectedRegistry: RegistryTreeItemBase | undefined;
 
-    let fullTag: string = node.fullTag;
-    if (fullTag.includes('/')) {
-        if (!defaultRegistryPath) {
-            await askToSaveRegistryPath(fullTag);
+    if (!node.fullTag.includes('/')) {
+        // The registry to push to is indeterminate--could be Docker Hub, or could need tagging.
+        const prompt: boolean = vscode.workspace.getConfiguration('docker').get('promptForRegistryWhenPushingImages', true);
+
+        // If the prompt setting is true, we'll ask; if not we'll assume Docker Hub.
+        if (prompt) {
+            connectedRegistry = await ext.registriesTree.showTreeItemPicker<RegistryTreeItemBase>(registryExpectedContextValues.all.registry, context);
```

Comment:
From Ravi--if you do "Connect Registry..." and give wrong credentials, you'll end up with the "No matching resources found" error.